src/test/compile-fail/private-in-public-warn.rs

```diff
@@ -246,6 +245,8 @@ mod aliases_priv {
     use self::Priv1 as PrivUseAlias;
     use self::PrivTr1 as PrivUseAliasTr;
     type PrivAlias = Priv2;
+    //~^ WARN private type in public interface
+    //~| WARNING hard error
```

Comment:
>Ideally, there should be two diagnostic messages - one for the public interface location and the other for actual private type location.

I implemented the extra note for the private type location for type aliases, but dropped it, the situation should be rare, likely happens in advanced code and doesn't worth the extra code.
Now the error is reported at the public interface location only.